### PR TITLE
Upgrade remaining compile options to lazy API

### PR DESCRIPTION
**Pull request by asodja**

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test AST

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
@bot-gradle test this

File: platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy [hunk 1]
```diff
@@ -55,9 +55,11 @@ abstract class CrossTaskIncrementalJavaCompilationIntegrationTest extends Abstra
         """
         file("impl/build.gradle") << """
             def layout = project.layout
-            compileJava.doFirst {
-                options.compilerArgs << "--module-path" << classpath.join(File.pathSeparator)
-                classpath = layout.files()
+            compileJava {
+                options.compilerArgs.addAll(["--module-path", classpath.join(File.pathSeparator)])
```
Comment (big-guy) [inline platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy:59]: This is not the same as it resolves classpath earlier

File: platforms/jvm/language-java/src/test/groovy/org/gradle/api/tasks/compile/CompileOptionsTest.groovy [hunk 2]
```diff
@@ -80,7 +80,7 @@ class CompileOptionsTest extends Specification {
 
     def "define"() {
         compileOptions.debug.set(false)
-        compileOptions.fork = false
+        compileOptions.fork.set(false)
```
Comment (bamboo) [inline platforms/jvm/language-java/src/test/groovy/org/gradle/api/tasks/compile/CompileOptionsTest.groovy:83]: ❓ Is `set` needed here?

File: platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy (hunk 1 above)
Comment (asodja) [reply platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy:59]: The problem is, that `ListProperty` is finalized and can't be modifed at the execution time. 
I think JavaCompile behavior stays the same even if we configure that earlier, but I need to double check it

File: platforms/jvm/language-java/src/test/groovy/org/gradle/api/tasks/compile/CompileOptionsTest.groovy (hunk 2 above)
Comment (asodja) [reply platforms/jvm/language-java/src/test/groovy/org/gradle/api/tasks/compile/CompileOptionsTest.groovy:83]: It actually works with `=`, true. It's just not detected by IDE

**Comment by asodja** [comment]
This was cherry-picked manually to https://github.com/gradle/gradle/pull/25963

**Pull request closed without merging**